ddtrace/tracer/option.go

```diff
@@ -125,14 +113,35 @@ func defaults(c *config) {
 			}
 			kv := strings.SplitN(tag, ":", 2)
 			k := strings.TrimSpace(kv[0])
-			switch len(kv) {
-			case 1:
-				WithGlobalTag(k, "")(c)
-			case 2:
-				WithGlobalTag(k, strings.TrimSpace(kv[1]))(c)
+			var v string
+			if len(kv) == 2 {
+				v = strings.TrimSpace(kv[1])
+			}
+			switch k {
+			case "service":
+				c.serviceName = v
+				globalconfig.SetServiceName(v)
+			case "env":
+				WithEnv(v)(c)
+			case "version":
+				c.version = v
+			default:
+				WithGlobalTag(k, v)(c)
 			}
 		}
 	}
+	if v := os.Getenv("DD_ENV"); v != "" {
```

Comment:
Since the global tags are a more "granular" type of setting referring to span rather than the whole app, should we not have them take precedence over the env. vars?

Reply:
The spec doc says `DD_ENV`, etc. take precedence over global tags.